redis/src/cluster_topology.rs

```diff
@@ -263,66 +255,80 @@ pub(crate) fn calculate_topology(
+        None => {
+            return Err(RedisError::from((
+                ErrorKind::ResponseError,
+                "No topology views found",
+            )));
+        }
+    };
     // Find the most frequent topology view
-    for curr_view in vec_iter {
-        let max_view = match most_frequent_topology {
-            Some(view) => view,
-            None => {
-                most_frequent_topology = Some(curr_view);
-                continue;
-            }
-        };
-        match max_view.cmp(curr_view) {
+    for mut curr_view in vec_iter {
+        match most_frequent_topology
+            .nodes_count
+            .cmp(&curr_view.nodes_count)
+        {
             std::cmp::Ordering::Less => {
-                most_frequent_topology = Some(curr_view);
-                has_more_than_a_single_max = false;
+                most_frequent_topology = curr_view;
+                non_unique_max_node_count = false;
             }
-            std::cmp::Ordering::Equal => has_more_than_a_single_max = true,
             std::cmp::Ordering::Greater => continue,
+            std::cmp::Ordering::Equal => {
+                non_unique_max_node_count = true;
+
+                // We choose as the greater view the one with higher slot coverage.
+                most_frequent_topology.parse_and_count_slots(tls_mode);
+                if let Some((slot_count, _)) = most_frequent_topology.slots_and_count {
+                    curr_view.parse_and_count_slots(tls_mode);
+                    let curr_slot_count = curr_view
+                        .slots_and_count
+                        .as_ref()
+                        .map(|(slot_count, _)| *slot_count)
+                        .unwrap_or(0);
+
+                    if let std::cmp::Ordering::Less = slot_count.cmp(&curr_slot_count) {
+                        most_frequent_topology = curr_view;
+                    }
+                } else {
+                    most_frequent_topology = curr_view;
+                }
+            }
         }
     }
-    let most_frequent_topology = match most_frequent_topology {
-        Some(view) => view,
-        None => unreachable!(),
+
+    let parse_and_built_result = |mut most_frequent_topology: TopologyView| {
+        let mut new_slots = SlotMap::new();
+        most_frequent_topology.parse_and_count_slots(tls_mode);
+        let slots_data = most_frequent_topology
+            .slots_and_count
+            .map(|(_, slots)| slots)
+            .ok_or(RedisError::from((
+                ErrorKind::ResponseError,
+                "Failed to parse the slots on the majority view",
+            )))?;
+        build_slot_map(&mut new_slots, slots_data);
+        Ok(new_slots)
     };
-    if has_more_than_a_single_max {
+
+    if non_unique_max_node_count {
         // More than a single most frequent view was found
         // If we reached the last retry, or if we it's a 2-nodes cluster, we'll return all found topologies to be checked by the caller
```

Comment:
"we'll return all found topologies to be checked by the caller" isn't true anymore, can you change it to something like (in better english lol) "we'll return a topology view with the max node count that has the fullest slot coverage"?